ska-telescope/sdp-prototype
# Add Docker compose deployment for SDP Tango devices.

## Changes
- Adds compose files to deploy SDPMaster and SDPSubarray devices.
- Added readme with quick start guide to the docker-compose deploy folder.
- Updated SDPMaster and SDPSubarray device servers to
  register devices when started if needed.

## Testing instructions

See Quick start section of `README.md` in  `deploy/docker-compose` folder.

From the `deploy/docker-compose` folder:
- Run `make up` to start a tango facility (tangodb and databaseds containers)
- Run `make start sdp_master` to register and start the SDPMaster device server
- Run `make start sdp_subarray` to register and start the SDPMaster device server
- Run `make itango_prompt` to start an itango container and replace the current prompt with the itango shell inside this container.
- Check the list of registered devices with `lsdev` in the itango shell.
- Get a device proxy connection to the SDPMaster with `d = DeviceProxy('mid_sdp/elt/master')`
- Check the status of the SDPMaster with `d.status()`
